geth/common/profiling.go

```diff
@@ -0,0 +1,55 @@
+package common
```

Comment:
1. Why have you decided to put profiling into `geth/common`? In my opinion, it's unrelated to geth and it definitely deserves a separate package just because it's a perfectly separate piece of functionality.
2. Have you looked into `go-ethereum`'s profiling capabilities? AFAIK, it can do it perfectly well.